mithril-client-cli/src/commands/cardano_db/verify.rs

```diff
@@ -0,0 +1,409 @@
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use anyhow::{anyhow, Context};
+use chrono::Utc;
+use clap::Parser;
+use mithril_client::{
+    cardano_database_client::ImmutableFileRange,
+    common::{ImmutableFileNumber, MKProof, ProtocolMessage},
+    CardanoDatabaseSnapshot, Client, MessageBuilder, MithrilCertificate, MithrilResult,
+};
+use slog::{debug, warn, Logger};
+
+use crate::{
+    commands::{cardano_db::CardanoDbCommandsBackend, client_builder, SharedArgs},
+    configuration::{ConfigError, ConfigParameters, ConfigSource},
+    utils::{
+        self, CardanoDbUtils, ExpanderUtils, IndicatifFeedbackReceiver, ProgressOutputType,
+        ProgressPrinter,
+    },
+    CommandContext,
+};
+
+use super::download::shared_steps;
+
+/// Clap command to verify a Cardano db and its associated certificate.
+#[derive(Parser, Debug, Clone)]
+pub struct CardanoDbVerifyCommand {
+    #[arg(short, long, value_enum, default_value_t)]
```

Comment:
Maybe the default should be `v2` here as this command doesn't work on v1 ?
```suggestion
    #[arg(short, long, value_enum, default_value_t = CardanoDbCommandsBackend::v2)]
```